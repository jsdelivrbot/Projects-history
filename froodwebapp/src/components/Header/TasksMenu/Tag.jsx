import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Tag as AntdTag } from 'antd';

class Tag extends Component {
  handleRemoveTag = () => {
    this.props.onClose(this.props.id);
  }

  render() {
    const {
      id,
      children,
      color
    } = this.props;

    return (
      <AntdTag
        style={ {
          display: 'block',
          margin: '0.5rem 0'
        } }
        color={ color }
        id={ id }
        key={ id }
        onClose={ this.handleRemoveTag }
        closable
      >
        { children }
      </AntdTag>
    );
  }
}

Tag.propTypes = {
  children: PropTypes.string,
  id: PropTypes.string,
  color: PropTypes.string,
  onClose: PropTypes.func,
};

export default Tag;
