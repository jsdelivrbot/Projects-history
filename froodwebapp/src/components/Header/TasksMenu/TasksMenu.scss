.tasksMenu {
  position: absolute;
  right: 0;
  top: 10.5rem;
  height: calc(100vh - 12.5rem);
  z-index: 50;
  background-color: white;
  padding: 2rem;
  border-left: 1px solid #CCCCCC;
}