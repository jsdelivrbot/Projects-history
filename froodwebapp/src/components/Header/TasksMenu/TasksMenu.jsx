import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Card } from 'antd';
import { connect } from 'react-redux';
import { removeWarningNotification } from 'redux-base/actions';
import Tag from './Tag';
import styles from './TasksMenu.scss';

const mapStateToProps = state => ({
  user: state.login.user,
  warningNotifications: state.error.warningNotifications
});

const mapDispatchToProps = { removeWarningNotification };

export class TasksMenu extends PureComponent {
  handleSetSidebarOpen = () => {
    this.props.handleMenuOpen();
  }

  render() {
    const {
      user,
      isOpened,
      warningNotifications
    } = this.props;

    return (

      <div
        className={ styles.tasksMenu }
        style={ {
          display: isOpened && user ? 'block' : 'none'
        } }
      >
        <Card title="Warnings" style={ { minWidth: '20rem' } }>
          { warningNotifications.map(warn => (
            <Tag
              color="orange"
              id={ warn.id }
              key={ warn.id }
              onClose={ this.props.removeWarningNotification }
              closable
            >
              { warn.msg }
            </Tag>
          ))}
        </Card>
      </div>
    );
  }
}

TasksMenu.propTypes = {
  handleMenuOpen: PropTypes.func.isRequired,
  isOpened: PropTypes.bool,
  user: PropTypes.object,
  warningNotifications: PropTypes.array,
  removeWarningNotification: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(TasksMenu);
