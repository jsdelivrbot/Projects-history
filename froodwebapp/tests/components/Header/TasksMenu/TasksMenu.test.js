import React from 'react';
import { TasksMenu } from 'components/Header/TasksMenu/TasksMenu';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, it, expect } = global;

describe('TasksMenu', () => {
  it('captures snapshot of TasksMenu', () => {
    const props = {
      warningNotifications: [{
        id: '12',
        msg: 'ERROR'
      }, {
        id: '13',
        msg: 'SUCCESS'
      }],
      handleMenuOpen: jest.fn(),
      removeWarningNotification: jest.fn()
    };
    const snappedComponent = shallow(<TasksMenu { ...props } />);

    expect(toJson(snappedComponent)).toMatchSnapshot();
  });
});
