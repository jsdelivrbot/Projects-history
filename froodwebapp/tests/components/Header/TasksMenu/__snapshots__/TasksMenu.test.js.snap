// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksMenu captures snapshot of TasksMenu 1`] = `
<div
  className="tasksMenu"
  style={
    Object {
      "display": "none",
    }
  }
>
  <Card
    style={
      Object {
        "minWidth": "20rem",
      }
    }
    title="Warnings"
  >
    <Tag
      closable={true}
      color="orange"
      id="12"
      key="12"
      onClose={[Function]}
    >
      ERROR
    </Tag>
    <Tag
      closable={true}
      color="orange"
      id="13"
      key="13"
      onClose={[Function]}
    >
      SUCCESS
    </Tag>
  </Card>
</div>
`;
